dev/archery/archery/compat.py

```diff
@@ -49,3 +50,10 @@ def _stringify_path(path):
             return str(path)
 
     raise TypeError("not a path-like object")
+
+
+def _import_pandas():
+    # ARROW-13425: avoid importing PyArrow from Pandas
+    sys.modules['pyarrow'] = None
```

Comment:
Should we unset this one after pandas has been correctly imported to leave open the possibility for other commands to try import pyarrow? I guess it might just reintroduce the bug again somewhere else and it might just make sense to explicitly document somewhere that `archery` can't depend on `arrow`.

Reply:
I have no idea what would be most desirable in this case. I would ask @jorisvandenbossche but he's on vacation :-)

Reply:
Sorry, also not directly an idea. Setting this to None generally makes pyarrow non-importable? (also for other packages that might get imported after pandas?) 
In general archery shouldn't use (directly or indirectly) pyarrow, so I think either way is fine.

Reply:
Ok, thank you.